### PR TITLE
Update Migration Guide

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Update method signature style of the `V6_MIGRATION` guide to match the Apple method signature syntax (example [here](https://developer.apple.com/documentation/passkit/pkpaymentauthorizationcontroller/1649455-canmakepayments))
- Other general cleanup

### Checklist

- ~[ ] Added a changelog entry~

- @jaxdesmarais

### CODE REVIEW

File: V6_MIGRATION.md [hunk 1]
```diff
@@ -76,9 +76,9 @@ venmoClient.tokenize(venmoRequest) { venmoAccountNonce, error in
 
 The property `BTPayPalRequest.activeWindow` has been removed
 
-Removed `BTPayPalDriver.requestOneTimePayment` and `BTPayPalDriver.requestBillingAgreement` in favor of `BTPayPalClient.tokenizePayPalAccount`:
+Removed `BTPayPalDriver.requestOneTimePayment` and `BTPayPalDriver.requestBillingAgreement` in favor of `BTPayPalClient.tokenize`:
 ```
-let payPalClient = BTPayPalClient(apiClient: <MY_BTAPICLIENT>)
+let payPalClient = BTPayPalClient(apiClient: <MY_BTAPIClient>)
```
Comment (jaxdesmarais) [inline V6_MIGRATION.md:81]: Updated this but not totally sure what our preferred syntax is for this. All caps? Just `<BTAPIClient>`? Something else entirely? Hoping someone has a strong preference and we can update these all accordingly.

File: V6_MIGRATION.md [hunk 2]
```diff
@@ -149,23 +148,22 @@ The `BraintreeUnionPay` module, and all containing classes, was removed in v6. U
...
-let cardClient = BTCardClient(apiClient: braintreeClient)
+let cardClient = BTCardClient(apiClient: <MY_BTAPIClient>)
 
 let card = BTCard()
 card.number = "4111111111111111"
 card.expirationMonth = "12"
 card.expirationYear = "2025"
 
-cardClient.tokenizeCard(card) { (tokenizedCard, error) in
+cardClient.tokenize(card) { tokenizedCard, error in
```
Comment (jaxdesmarais) [inline V6_MIGRATION.md:158]: Below this we have `// Communicate the tokenizedCard.nonce to your server, or handle error` but in the rest of this file we just do `// handle response`. Do folks have a preference on this? We should update them all to be the same, imo.

File: V6_MIGRATION.md (hunk 1 above)
Comment (sshropshire) [reply V6_MIGRATION.md:81]: Could we instantiate a BTAPIClient with a "<TOKENIZATION_KEY_OR_CLIENT_TOKEN>" auth string? Not sure if that's too much but it'll have a similar effect as above and be valid for the copy/paste.

**Review by sshropshire — APPROVED** [review]

File: V6_MIGRATION.md (hunk 1 above)
Comment (scannillo) [reply V6_MIGRATION.md:81]: I think in general the `<SOME_VALUE>` syntax is all-caps, and is usually for strings.

I would do something like:
```
let apiClient = BTAPIClient("<TOKENIZATION_KEY_OR_CLIENT_TOKEN>")
let payPalClient = BTPayPalClient(apiClient: apiClient)
```

Comment (scannillo) [reply V6_MIGRATION.md:81]: And then in the examples below, just use `apiClient` as the param.

File: V6_MIGRATION.md [hunk 3]
```diff
@@ -54,7 +54,7 @@ The possible values for `BTVenmoPaymentMethodUsage` include:
 * `.multiUse` - the Venmo payment will be authorized for future payments and can be vaulted.
 * `.singleUse` - the Venmo payment will be authorized for a one-time payment and cannot be vaulted.
 
-`BTVenmoClient.tokenizeVenmoAccount(with: BTVenmoRequest, completion: @escaping (BTVenmoAccountNonce?, Error?) -> Void)` has been renamed to `BTVenmoClient.tokenize(_ request: BTVenmoRequest, completion: @escaping (BTVenmoAccountNonce?, Error?) -> Void)`
+`BTVenmoClient.tokenizeVenmoAccount(with:completion:)` has been renamed to `BTVenmoClient.tokenize(_:completion:)`
```
Comment (scannillo) [inline V6_MIGRATION.md:57]: This makes things a lot easier to read, too! We could also audit the CHANGELOG entries for the beta cleaning these up too.

Comment (jaxdesmarais) [reply V6_MIGRATION.md:57]: I looked through the CHANGELOG entries and wasn't able to find anywhere where we had the other format, it looked like so far we've always used this syntax there.

File: V6_MIGRATION.md (hunk 1 above)
Comment (jaxdesmarais) [reply V6_MIGRATION.md:81]: Dope! I'll update these - I dig that too.

File: V6_MIGRATION.md (hunk 2 above)
Comment (jaxdesmarais) [reply V6_MIGRATION.md:167]: Updated to `handle response`: https://github.com/braintree/braintree_ios/pull/984/commits/5842ae352f8324be34cc5c372319765e22a2aa72

File: V6_MIGRATION.md (hunk 1 above)
Comment (jaxdesmarais) [reply V6_MIGRATION.md]: Done: https://github.com/braintree/braintree_ios/pull/984/commits/5842ae352f8324be34cc5c372319765e22a2aa72

**Review by KunJeongPark — APPROVED** [review]

**Pull request merged**